### ISSUE TITLE
Support unreachable blocks in partially defined check

### ISSUE THREAD

**Issue by ilinum** [python/mypy#13929]
For the partially defined check (`--enable-error-code partially-defined`), mypy should detect unreachable blocks:

Assuming we're running python3.6 in the following block:
```python
if sys.version_info >= (3, 6):
    x = 1
print(x)  # This should not generate an error but currently does.
```
